pkg/proxy/ipvs/proxier.go

```diff
@@ -922,13 +926,26 @@ func (proxier *Proxier) syncProxyRules() {
 	// We assume that if this was called, we really want to sync them,
 	// even if nothing changed in the meantime. In other words, callers are
 	// responsible for detecting no-op changes and not calling this function.
+
+	// (khenidak) the previous assumption is valid only for single stack
+	// for dual stack, we have no way for endpointslice (or general endpoint processing)
+	// to know if this endpoint is being processed for an IPv6 or IPv4 proxier
+	// TODO: Find a way where the endpoints/slice can be loaded with knowledge
+	// of proxy/ipfamily and perform the filtering there.
 	serviceUpdateResult := proxy.UpdateServiceMap(proxier.serviceMap, proxier.serviceChanges)
 	endpointUpdateResult := proxier.endpointsMap.Update(proxier.endpointsChanges)
 
 	staleServices := serviceUpdateResult.UDPStaleClusterIP
 	// merge stale services gathered from updateEndpointsMap
 	for _, svcPortName := range endpointUpdateResult.StaleServiceNames {
 		if svcInfo, ok := proxier.serviceMap[svcPortName]; ok && svcInfo != nil && svcInfo.Protocol() == v1.ProtocolUDP {
+			// ignore stale services that does not match ipfamily of the proxier
+			isIpv6Svc := utilnet.IsIPv6(svcInfo.ClusterIP())
```

Comment:
What about ExternalIP?

Reply:
A service that has `ExternalIPs` must have `ClusterIP` AFAIK. Do you know of any case where a service can carry `ExternalIPs` but `ClusterIP` is empty?

Reply:
Headless Services (ClusterIP: None) with external IPs?

Reply:
```
apiVersion: v1
kind: Service
metadata:
  name: nginx
  labels:
    app: nginx
spec:
  clusterIP: None
  externalIPs:
  - 1.1.1.1
  selector:
    app: nginx
  ports:
  - port: 80
    protocol: TCP
```

```
$ kubectl get svc nginx
NAME    TYPE        CLUSTER-IP   EXTERNAL-IP   PORT(S)   AGE
nginx   ClusterIP   None         1.1.1.1       80/TCP    62s
```